chain/client/src/view_client.rs

```diff
@@ -268,22 +418,22 @@ impl ViewClientActor {
             QueryRequest::CallFunction { account_id, .. } => account_id,
             QueryRequest::ViewCode { account_id, .. } => account_id,
         };
-        let shard_id =
-            self.runtime_adapter
-                .account_id_to_shard_id(account_id, header.epoch_id())
-                .map_err(|err| QueryError::InternalError { error_message: err.to_string() })?;
-        let shard_uid = self
-            .runtime_adapter
+        let runtime_adapter = self.runtime_adapter(temp);
+        let shard_id = runtime_adapter
```

Comment:
If we query something for epoch_id, should we use runtime_adapter of the epoch_id temperature (which is not  always header temperature)?

Reply:
It is a good question and honestly I don’t know conclusively.  It may be that temperature and gc will need to take epoch boundaries into account.  For a block to be hot or cold, the entire epoch may need to be hot or cold.  But then again, if all we need is epoch info than, as you’ve observed below, it may be enough to just read EpochInfo from hot and deal with the problem if that column becomes garbage collected.